nix/shell.nix

```diff
@@ -0,0 +1,34 @@
+{ pkgs, esy ? pkgs.nodePackages.esy, deku, npmPackages }:
```

Comment:
On Mac M1:
```shell script
error: Package ‘ocaml4.12.1-ligo-0.34.0’ in /nix/store/mxvjplj9qxxnximgq8ljwdjmjjalnbfb-source/pkgs/development/compilers/ligo/default.nix:60 is not supported on ‘aarch64-darwin’, refusing to evaluate.
```
I don't know where it is coming from, can be by-passed by running
```shell script
# For `nix shell`, `nix build`, `nix develop` or any other Nix 2.4+
# (Flake) command, `--impure` must be passed in order to read this
# environment variable.
nix develop --impure -c $SHELL
```

Reply:
Which, finally, leads to:
```shell script
error: builder for '/nix/store/rsdh992kjck697l4nmf6z3lnbid3z9hn-ocaml4.12.1-hacl-star-raw-0.4.5.drv' failed with exit code 2;
       last 10 log lines:
       >  does not support x64 assembly, disabling Curve64
       >  does not support legacy vale stubs
       >  does not support 256-bit arithmetic
       > ... found ocamlfind in /nix/store/cl0x22l6h1q93vzkzv144h39pvaajrj6-ocaml-findlib-1.9.1/bin/ocamlfind
       > ... found ctypes in /nix/store/fs8dxasrkq15ynyz5qi65hr955zx30cn-ocaml4.12.1-ctypes-0.18.0/lib/ocaml/4.12.1/site-lib/ctypes
       > building
       > build flags: SHELL=/nix/store/m0nsx4ai768kjp9c40vhkpj9pxlii46m-bash-5.1-p12/bin/bash
       > clang-11: error: argument unused during compilation: '-fno-strict-overflow' [-Werror,-Wunused-command-line-argument]
       > Makefile.basic:53: Hacl_AES.d: No such file or directory
       > make: *** [Makefile.basic:48: Hacl_AES.d] Error 1
       For full logs, run 'nix log /nix/store/rsdh992kjck697l4nmf6z3lnbid3z9hn-ocaml4.12.1-hacl-star-raw-0.4.5.drv'.
error: 1 dependencies of derivation '/nix/store/20g8wl79qgd9r9b4vvkdgl7v8cbk6qin-ocaml4.12.1-hacl-star-0.4.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/7zpp49319wl1igr2zbzcrzmd6dn458ay-ocaml4.12.1-tezos-hacl-glue-unix-11.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qgr0ykc8xmbxin74aw9srasf6jqagwxb-ocaml4.12.1-tezos-base-11.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qy7ahm2yygmbndhmanncjg4lbg7rnar8-ocaml4.12.1-tezos-crypto-11.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5g49s3v0gvlf6zhfraja5jp20b8snfkc-ocaml4.12.1-ligo-0.34.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5bpx30h3jgmlrrl182nl2dmlc2yh1fb2-nix-shell-env.drv' failed to build
```

Reply:
Maybe we can remove LIGO for now? Or only add it on Linux?

Reply:
We're using the docker image for Ligo in the scripts. I only use this for debugging, but I'll remove and just do `nix-shell -p ligo` whenever I need Ligo locally.

Reply:
done